From fa9ffde8a54eccb611f189b574afd7fb0141c061 Mon Sep 17 00:00:00 2001
From: Patrick O'Neill <patrick@rivosinc.com>
Date: Fri, 1 Dec 2023 15:56:14 -0800
Subject: [PATCH 3/3] Sidestep implicitly defined function headers

---
 newlib/libc/include/math.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/newlib/libc/include/math.h b/newlib/libc/include/math.h
index 54e30ef82..21f5ffa7c 100644
--- a/newlib/libc/include/math.h
+++ b/newlib/libc/include/math.h
@@ -445,7 +445,6 @@ extern float hypotf (float, float);
    simply call the double functions.  On Cygwin the long double functions
    are implemented independently from newlib to be able to use optimized
    assembler functions despite using the Microsoft x86_64 ABI. */
-#if defined (_LDBL_EQ_DBL) || defined (__CYGWIN__)
 /* Reentrant ANSI C functions.  */
 #ifndef __math_68881
 extern long double atanl (long double);
@@ -515,7 +514,6 @@ extern long double lgammal (long double);
 extern long double erfl (long double);
 extern long double erfcl (long double);
 #endif /* ! defined (_REENT_ONLY) */
-#else /* !_LDBL_EQ_DBL && !__CYGWIN__ */
 extern long double hypotl (long double, long double);
 extern long double sqrtl (long double);
 extern long double frexpl (long double, int *);
@@ -525,7 +523,6 @@ extern _LONG_DOUBLE rintl (_LONG_DOUBLE);
 extern long int lrintl (_LONG_DOUBLE);
 extern long long int llrintl (_LONG_DOUBLE);
 #endif /* __i386__ */
-#endif /* !_LDBL_EQ_DBL && !__CYGWIN__ */
 
 #endif /* __ISO_C_VISIBLE >= 1999 */
 
-- 
2.34.1

